lib/Dialect/Torch/Transforms/RefinePublicReturn.cpp

```diff
@@ -55,25 +74,22 @@ class RefinePublicReturnPass
       return signalPassFailure();
     }
 
-    // Get the new operands. Either the original operand, or if there is a
-    // TensorStaticInfoCastOp then the pre-casted operand, which is presumed to
-    // have a more precise type.
     SmallVector<Value> newOperands;
     OpBuilder builder(returnOp);
-    for (auto operand : returnOp.getOperands()) {
-      Value newOperand;
-      if (auto cast = operand.getDefiningOp<TensorStaticInfoCastOp>()) {
-        newOperand = cast.getOperand();
-      } else {
-        newOperand = operand;
-      }
-      if (auto tensorType = newOperand.getType().dyn_cast<BaseTensorType>()) {
-        newOperands.push_back(
-            copyTensorToType(builder, returnOp->getLoc(),
-                             tensorType.getWithValueSemantics(), newOperand));
-      } else {
-        newOperands.push_back(newOperand);
+    // There should always be just one return value. Multiple returns values are
```

Comment:
I don't think this is the right approach. It is just undoing what we did in AdjustCallingConventions :)

Why did you find you needed this?

Reply:
Oh, maybe I should handle this in `AdjustCallingConventions` instead? Tuple return value currently is not expanded to multiple values.

Reply:
Ah, lol. Yeah, that looks like it never got implemented.